Wrappers/Python/cil/optimisation/operators/BlockOperator.py

```diff
@@ -191,41 +195,46 @@ def direct(self, x, out=None):
             x_b = BlockDataContainer(x)
         else:
             x_b = x
-        shape = self.get_output_shape(x_b.shape)        
+            
+        if x_b.shape != self._domain_block_shape:
+            raise ValueError(
+                'We expect the input to be a block data container of shape {}'.format(  self._domain_block_shape))
+        
+        return_data_container = False
+        
+        if self._range_block_shape[0]==1:
```

Comment:
```suggestion
        if self._range_block_shape == (1,1):
```

This may be clearer to read in some time when we've forgotten what this code is about